packages/ws-response/index.js

```diff
@@ -0,0 +1,62 @@
+import ApiGatewayManagementApi from 'aws-sdk/clients/apigatewaymanagementapi.js'// v2
+// import {ApiGatewayManagementApi} from '@aws-sdk/client-apigatewaymanagementapi' // v3
+
+import { canPrefetch, createClient, createPrefetchClient } from '@middy/util'
+
+const defaults = {
+  AwsClient: ApiGatewayManagementApi,
+  awsClientOptions: {}, // { endpoint }
+  awsClientAssumeRole: undefined,
+  awsClientCapture: undefined,
+  disablePrefetch: false
+}
+
+const wsResponderMiddleware = (opts) => {
+  const options = { ...defaults, ...opts }
+
+  let client
+  if (canPrefetch(options) && options.awsClientOptions.endpoint) {
+    client = createPrefetchClient(options)
+  }
+
+  const wsResponderMiddlewareAfter = async (request) => {
+    normalizeWsResponse(request)
+    const { response } = request
+
+    if (!response.ConnectionId) return
+
+    if (!options.awsClientOptions.endpoint && request.event.requestContext) {
+      options.awsClientOptions.endpoint = request.event.requestContext.domainName + '/' + request.event.requestContext.stage
+    }
+    if (!client) {
+      client = await createClient(options, request)
+    }
+
+    await client
+      .postToConnection(response)
+      .promise()
+
+    request.response = {
+      statusCode: 200
+    }
+  }
+
+  return {
+    after: wsResponderMiddlewareAfter
+  }
+}
+
+// TODO move to @middy/util?
```

Comment:
I would consider this specific to websockets (therefore relevant only for this middleware). Or are you anticipating users might want to use this even without the middleware?

Reply:
Because there is only one `ws` middleware, we don't need it in `util`. But can move if future use cases arise. I hope to be able to use this myself later in the year.